ddtrace/settings.py

```diff
@@ -69,11 +70,109 @@ def _add(self, integration, settings, merge=True):
             # >>> config._add('requests', dict(split_by_domain=False))
             # >>> config.requests['split_by_domain']
             # True
-            self._config[integration] = deepmerge(existing, settings)
+            self._config[integration] = IntegrationConfig(self, deepmerge(existing, settings))
         else:
-            self._config[integration] = settings
+            self._config[integration] = IntegrationConfig(self, settings)
+
+    def trace_headers(self, whitelist):
+        """
+        Registers a set of headers to be traced at global level or integration level.
+        :param whitelist: the case-insensitive list of traced headers
+        :type whitelist: list of str or str
+        :return: self
+        :rtype: HttpConfig
+        """
+        self._http.trace_headers(whitelist)
+        return self
+
+    def header_is_traced(self, header_name):
+        """
+        Returns whether or not the current header should be traced.
+        :param header_name: the header name
+        :type header_name: str
+        :rtype: bool
+        """
+        return self._http.header_is_traced(header_name)
 
     def __repr__(self):
         cls = self.__class__
         integrations = ', '.join(self._config.keys())
         return '{}.{}({})'.format(cls.__module__, cls.__name__, integrations)
+
+
+class IntegrationConfig(dict):
+
+    def __init__(self, global_config, *args, **kwargs):
+        """
+        :param global_config:
+        :type global_config: Config
+        :param args:
+        :param kwargs:
+        """
+        super(IntegrationConfig, self).__init__(*args, **kwargs)
+        self.http = HttpConfig()
+        self.global_config = global_config
```

Comment:
I think my only complaint here is if we move this to be an `AttrDict`, then `.http` and `.global_config` will become reserved names.

wonder if we just want to do `self.http_config` to make it less likely?

@Kyle-Verhoog ❓